langchain/docstore/base.py

```diff
@@ -23,3 +39,135 @@ class AddableMixin(ABC):
     @abstractmethod
     def add(self, texts: Dict[str, Document]) -> None:
         """Add more documents."""
+
+
+class DocManager(ABC, Generic[UID_TYPE]):
```

Comment:
We could explore providing a persistence layer for prototyping purposes that behaves as the doc manager. Say could help users persist the documents on disk, so they don't have to re-generate them. The understanding would be that this persistence layer is mostly for prototyping purposes since it won't scale well. 

If we do that, I think it makes sense to de-couple it from any vectorstores. I like your suggestion from the  other pr of staying with the convention of explicitly providing the `ids` as part of the method signature.

```python
def add_texts(..., ids: Optional[List[str]]=None):
  for i, t in enumerate(text):
    id = ids[i] if ids else str(uuid4())
```